plugins/base/src/main/kotlin/transformers/pages/DefaultSamplesTransformer.kt

```diff
@@ -51,14 +52,14 @@ internal class DefaultSamplesTransformer(val context: DokkaContext) : PageTransf
     private fun ContentNode.addSample(
         contentPage: ContentPage,
         fqLink: String,
-        sample: SampleProvider.SampleSnippet,
+        sample: SampleSnippet,
     ): ContentNode {
         val node = contentCode(contentPage.content.sourceSets, contentPage.dri, createSampleBody(sample.imports, sample.body), "kotlin")
         return dfs(fqLink, node)
     }
 
-    fun createSampleBody(imports: String, body: String) =
-        """ |$imports
+    private fun createSampleBody(imports: List<String>, body: String) =
```

Comment:
> why do we need .takeIf { it.isNotEmpty() } here? if list is empty, message from joinToString will be empty

Left a comment in https://github.com/Kotlin/dokka/commit/ea88a0a6e6666b31b4a4071f77c80889ac22d9af. TLDR is that if `imports` are empty, there should be no empty/blank lines. But if `imports` are present, there should be an empty line between them and the body.

It's very hacky, and I want to refactor this whole class, but this PR is not small as it is, so I tried to avoid unnecessary changes. Hopefully it'll be completely re-written in #3041 anyway

> Also, default separator in joinToString is , - it should be replaced with empty string or new line
> Looks like there is no tests for this :)

We have some tests (which is how I remember about this place and why I added the condition), but none of them have multiple import lines :) So you indeed found a bug. I've changed one of the tests to check it now, but again, hopefully better tests will be added as part of #3041

Reply:
Thanks, now I see the difference. It will be good then to add comment to #3041 about additional tasks to follow-up

Reply:
Added it to the issue's description that it's expected to have the refactoring and that the tests will need to be revised